test/async-hooks/test-ttywrap.readstream.js

```diff
@@ -1,32 +1,42 @@
 'use strict';
+
 const common = require('../common');
 const assert = require('assert');
-
-// general hook test setup
 const tick = require('./tick');
 const initHooks = require('./init-hooks');
 const { checkInvocations } = require('./hook-checks');
+const tty_fd = common.getTTYfd();
+
+if (tty_fd < 0)
+  return common.skip('no valid TTY fd available');
+const ttyStream = (() => {
+  try {
+    return new (require('tty').ReadStream)(tty_fd);
```

Comment:
Hm. Reason I `return null` here then check it below is because I'd have imagined that `common.getTTYfd()` should only return a positive number if there's a valid fd for the TTY. Thus `ReadStream`/`WriteStream` shouldn't fail. Appears that isn't always the case. So I'll go back to the `null` check and early return.

I will defend the use of `getTTYfd()`, which also does a `require('tty')` when called. So I'm not worried about the possible side effects of calling it here.